### PR TITLE
New Terms of Service; Didn't Read Spice

### PR DESCRIPTION
**Pull request by javathunderman**
## Description of new Instant Answer, or changes
Creates a Terms of Service; Didn't Read spice. Fetches and displays API data from tosdr.org on company terms of services and privacy policies. 

## Related Issues and Discussions
Currently has only very basic functionality. The JSON response generated by the TOS;DR API has kind of weird formatting, and it's a little hard to parse. Also not sure what other information I should add to this IA. 

## People to notify
@duckduckgo/community-leaders 

Instant Answer Page: https://duck.co/ia/view/terms_of_service_didn_t_read

It's good to be back.

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by J-tt** [comment]
This is very confusing to use in my opinion, and at first it felt broken.

When you search all that is returned is a Class.

(EG: "Google terms of service" returns the text "Class C")

I believe if some subtext was added, either describing the class or alternatively describing what the TOSDR service actually is it would hugely improve the users experience.

**Comment by javathunderman** [comment]
@J-tt sorry! I knew this was a little half-baked. Definitely will add the subtext. I'm trying to scrape some more content off of the API response to display as well.

**Comment by J-tt** [comment]
@javathunderman  No worries, just wanted to help improve what I think is an awesome addition to duckduckgo, definitely something I know I will use 👍

File: share/spice/terms_of_service/terms_of_service.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+            return Spice.failed('terms_of_service');
+        }
+
+        Spice.add({
+            id: "terms_of_service",
+            name: "Terms of Service",
+            data: api_result,
+            meta: {
+                sourceName: "Terms of Service; Didn't Read",
+                sourceUrl: 'http://tosdr.org'
```
Comment (TiagoDanin) [inline share/spice/terms_of_service/terms_of_service.js]: Why not use HTTPS here ?

Comment (javathunderman) [inline share/spice/terms_of_service/terms_of_service.js]: Typo, will fix that.

**Comment by javathunderman** [comment]
Anyone know a way to scrape the "tldr" value off of [this](https://tosdr.org/api/1/service/duckduckgo.json) JSON? Not sure how to do this without causing a bunch of errors...

**Comment by J-tt** [comment]
@javathunderman 

I made a jsfiddle that scrapes the data, the important part is:

```javascript
console.log(data.pointsData[data.points[0]].tosdr.tldr)
```

In that example we assume that `data` is the full api response.

[here](https://jsfiddle.net/re4eqwf0/) is the fiddle, please excuse the synchronous call, it's just a quick dirty example.

**Comment by javathunderman** [comment]
@J-tt Thanks! Seems to work perfectly.

**Comment by javathunderman** [comment]
This IA seems to be working alright, but I can't seem to remove the HTML contained in the JSON with `strip_html()`

**Comment by J-tt** [comment]
@javathunderman You could try: 

```javascript
var StrippedString = OriginalString.replace(/(<([^>]+)>)/ig,"");
```

Source: https://css-tricks.com/snippets/javascript/strip-html-tags-in-javascript/

**Comment by javathunderman** [comment]
@J-tt is there a good way to filter out companies/sites not contained in the API? I thought about using the handle function and a file in the share directory, but I'm wondering if there's a better way to do it.

**Comment by javathunderman** [comment]
Due to [this](https://forum.duckduckhack.com/t/duckduckhack-is-now-in-maintenance-mode/2559) announcement, I'm closing this PR.

**Comment by javathunderman** [comment]
@pjhampton I'm open to finishing this if needed.
